ekncontent/ekncontent/eknc-engine.c

```diff
@@ -480,14 +473,23 @@ eknc_engine_get_domain_for_app (EkncEngine *self,
   g_return_val_if_fail (app_id && *app_id, NULL);
   g_return_val_if_fail (error == NULL || *error == NULL, NULL);
 
-  if (g_hash_table_contains (self->domains, app_id))
-    return g_hash_table_lookup (self->domains, app_id);
+  EkncDomain *domain = g_hash_table_lookup (self->domains, app_id);
 
-  EkncDomain *domain;
-  if (!(domain = eknc_domain_get_impl (app_id, NULL, self->xapian_bridge, NULL, error)))
+  if (domain != NULL)
+    return domain;
+
+  domain = eknc_domain_get_for_app_id (app_id, NULL, self->language, NULL, error);
+  if (domain == NULL)
     return NULL;
+
+  // Bind our own language property to the domain
+  g_object_bind_property (self, "language",
```

Comment:
This doesn't work (and isn't necessary) now that the property is construct-only.